src/server/scripts/Kalimdor/zone_dustwallow_marsh.cpp

```diff
@@ -15,215 +15,104 @@
-                return ValidateSpellInfo({ SPELL_OOZE_ZAP });
-            }
-
-            SpellCastResult CheckRequirement()
-            {
-                if (!GetCaster()->HasAura(GetSpellInfo()->Effects[EFFECT_1].CalcValue()))
-                    return SPELL_FAILED_CANT_DO_THAT_RIGHT_NOW; // This is actually correct
-
-                if (!GetExplTargetUnit())
-                    return SPELL_FAILED_BAD_TARGETS;
-
-                return SPELL_CAST_OK;
-            }
-
-            void HandleDummy(SpellEffIndex effIndex)
-            {
-                PreventHitDefaultEffect(effIndex);
-                if (GetHitUnit())
-                    GetCaster()->CastSpell(GetHitUnit(), uint32(GetEffectValue()), true);
-            }
-
-            void Register() override
-            {
-                OnEffectHitTarget += SpellEffectFn(spell_ooze_zap_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
-                OnCheckCast += SpellCheckCastFn(spell_ooze_zap_SpellScript::CheckRequirement);
-            }
-        };
-
-        SpellScript* GetSpellScript() const override
-        {
-            return new spell_ooze_zap_SpellScript();
-        }
+    void HandleDummy(SpellEffIndex effIndex)
+    {
+        PreventHitDefaultEffect(effIndex);
+        if (Player* player = GetCaster()->ToPlayer())
+            player->CastSpell(player, SPELL_OOZE_CHANNEL_CREDIT, true);
+        GetHitUnit()->KillSelf();
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_ooze_zap_channel_end::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
 };
 
-class spell_ooze_zap_channel_end : public SpellScriptLoader
+// 42492 - Cast Energized
+class spell_energize_aoe : public SpellScript
 {
-    public:
-        spell_ooze_zap_channel_end() : SpellScriptLoader("spell_ooze_zap_channel_end") { }
+    PrepareSpellScript(spell_energize_aoe);
 
-        class spell_ooze_zap_channel_end_SpellScript : public SpellScript
-        {
-            PrepareSpellScript(spell_ooze_zap_channel_end_SpellScript);
-
-            bool Validate(SpellInfo const* /*spellInfo*/) override
-            {
-                return ValidateSpellInfo({ SPELL_OOZE_ZAP_CHANNEL_END });
-            }
-
-            void HandleDummy(SpellEffIndex effIndex)
-            {
-                PreventHitDefaultEffect(effIndex);
-                if (Player* player = GetCaster()->ToPlayer())
-                    player->CastSpell(player, SPELL_OOZE_CHANNEL_CREDIT, true);
-                GetHitUnit()->KillSelf();
-            }
-
-            void Register() override
-            {
-                OnEffectHitTarget += SpellEffectFn(spell_ooze_zap_channel_end_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
-            }
-        };
-
-        SpellScript* GetSpellScript() const override
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        for (std::list<WorldObject*>::iterator itr = targets.begin(); itr != targets.end();)
```

Comment:
you can use auto for iterators

Reply:
Lol, no. You have the type literally in front of your face and it's a very basic one so no auto when there is no real reason to use it (like  type<type<type<type>>> insanity. We aint using C# here lul. Also that's not part of this PR. He's switching the scripts over to the register macro model, so cutting away the script loaders basically.

Reply:
its the standard here, auto is allowed for iterators.
about this just being a refactor, yeah I missed this code actually already existed, my bad, lets move along.